pkg/solana/txm/pendingtx.go

```diff
@@ -48,12 +51,13 @@ type PendingTxContext interface {
 
 // finishedTx is used to store info required to track transactions to finality or error
 type pendingTx struct {
-	tx         solana.Transaction
-	cfg        TxConfig
-	signatures []solana.Signature
-	id         string
-	createTs   time.Time
-	state      TxState
+	tx                   solana.Transaction
+	cfg                  TxConfig
+	signatures           []solana.Signature
+	id                   string
+	createTs             time.Time
+	state                TxState
+	lastValidBlockHeight uint64 // to track expiration
```

Comment:
In Solana, each transaction includes a `"recent blockhash"` that expires when that blockhash is no longer `"recent enough"`

During transaction processing, Solana validators will check if the `"recent blockhash"` of each transaction is recorded within the 151 most recent stored hashes. If the transaction's `"recent blockhash"` is older than this maximum processing age, the transaction won't be processed.

To track the expiration of each transaction, we call the RPC method [GetLatestBlockhash](https://solana.com/docs/rpc/http/getlatestblockhash) and store the return value `lastValidBlockHeight` within our `pendingTx` object. This value represents the height of the last block at which the returned `blockhash` value will be valid. 

Since slots (period of time a validator can produce a block) fluctuate between 400 ms and 600 ms, transactions can only use a given blockhash for 150 blocks, which is 60 to 90 seconds before the validators consider it expired.

Reply:
Ah yeah this became clear to me later in the PR review, thanks for clarifying though that makes sense